Warfare/Warfare/GameEngine.swift

```diff
@@ -17,6 +17,331 @@ class GameEngine {
         self.loadMap(number: "1")
     }
 
+    // MARK: - Operations
+
+
+    // Set up the gameState after which the player can start giving out orders
+    func beginTurn() {
+        for village in self.game.currentPlayer.villages {
+            for tile in village.controlledTiles {
+
+                // Replace tombstones
+                tile.replaceTombstone()
+
+                // Produce constructions and set unit actions.
+                // ReadyForOrders for all except first phase cultivation
```

Comment:
what about a master "if" : `if let action = tile.unit?.currentAction { }` and then use action?